cloud/google/gce_mig.py

```diff
@@ -0,0 +1,763 @@
+                      'ABANDONING', 'RESTARTING', 'REFRESHING']
+    instance_names = _get_instance_list(mig, filter_list=actions_filter)
+    if mig.destroy():
+        changed = True
+        return_data = instance_names
+
+    return (changed, return_data)
+
+
+def recreate_instances_in_mig(mig):
+    """
+    Recreate the instances for a Managed Instance Group.
+
+    :param mig: Managed Instance Group Object from libcloud.
+    :type mig:  :class: `GCEInstanceGroupManager`
+
+    :return: Tuple with changed stats and a list of affected instances.
+    :rtype: tuple in the format of (bool, list)
+    """
+    changed = False
+    return_data = []
+    actions_filter = ['RECREATING']
+
+    if mig.recreate_instances():
+        changed = True
+        return_data = _get_instance_list(mig, filter_list=actions_filter)
+
+    return (changed, return_data)
+
+
+def resize_mig(mig, size):
+    """
+    Resize a Managed Instance Group.
+
+    Based on the size provided, GCE will automatically create and delete
+    VMs as needed.
+
+    :param mig: Managed Instance Group Object from libcloud.
+    :type mig:  :class: `GCEInstanceGroupManager`
+
+    :return: Tuple with changed stats and a list of affected instances.
+    :rtype: tuple in the format of (bool, list)
+    """
+    changed = False
+    return_data = []
+    actions_filter = ['CREATING', 'DELETING']
+
+    if mig.resize(size):
+        changed = True
+        return_data = _get_instance_list(mig, filter_list=actions_filter)
+
+    return (changed, return_data)
+
+
+def get_mig(gce, name, zone):
+    """
+    Get a Managed Instance Group from GCE.
+
+    If the MIG is not found, None is found.
+
+    :param gce: An initialized GCE driver object.
+    :type gce:  :class: `GCENodeDriver`
+
+    :param name: Name of the Managed Instance Group.
+    :type name:  ``str``
+
+    :param zone: Zone that the Managed Instance Group is located in.
+    :type zone:  ``str``
+
+    :return: A GCEInstanceGroupManager object or None.
+    :rtype: :class: `GCEInstanceGroupManager` or None
+    """
+    try:
+        # Does the MIG already exist?
+        return gce.ex_get_instancegroupmanager(name=name, zone=zone)
+
+    except ResourceNotFoundError:
+        return None
+
+
+def main():
+
+    module = AnsibleModule(argument_spec=dict(
+        name=dict(required=True),
+        template=dict(),
+        recreate_instances=dict(type='bool', default=False),
+        # Do not set a default size here.  For Create and some update
+        # operations, it is required and should be explicitly set.
+        # Below, we set it to the existing value if it has not been set.
+        size=dict(type='int'),
+        state=dict(choices=['absent', 'present'], default='present'),
+        zone=dict(required=True),
+        autoscaling=dict(type='dict', default=None),
+        service_account_email=dict(),
+        service_account_permissions=dict(type='list'),
+        pem_file=dict(),
```

Comment:
It needs to be there, at least for the time-being.  The gce_connect function, defined in module_utils will look for these params (we also currently use the secrets file, a convention of libcloud, to authenticate, which is why they aren't required).  I suspect it's possible to define these once and then have all the modules leverage it.